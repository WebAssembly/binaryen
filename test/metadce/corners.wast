(module
 (import "env" "STACKTOP" (global $STACKTOP$asm2wasm$import i32))
 (global $STACKTOP (mut i32) (get_global $STACKTOP$asm2wasm$import))

 (import "env" "UNUSEDTOP" (global $UNUSEDTOP$asm2wasm$import i32))
 (global $UNUSEDTOP (mut i32) (get_global $UNUSEDTOP$asm2wasm$import))

 (import "env" "imported_twice" (func $imported_twice_a)) ;; and used just once,
 (import "env" "imported_twice" (func $imported_twice_b)) ;; but the other should not kill the import for both!

 (import "env" "an-imported-table-func" (func $imported_table_func))

 (import "env" "table" (table 10 10 anyfunc))
 (elem (i32.const 0) $imported_table_func)

 (export "stackAlloc" (func $stackAlloc))

 (func $stackAlloc
  (drop (get_global $STACKTOP))
  (call $imported_twice_a)
 )
)

