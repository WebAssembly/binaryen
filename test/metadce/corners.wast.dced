(module
 (type $FUNCSIG$v (func))
 (import "env" "table" (table $0 10 10 funcref))
 (elem (i32.const 0) $imported_table_func)
 (import "env" "STACKTOP" (global $STACKTOP$asm2wasm$import i32))
 (import "env" "imported_twice" (func $imported_twice_a))
 (import "env" "an-imported-table-func" (func $imported_table_func))
 (global $STACKTOP (mut i32) (global.get $STACKTOP$asm2wasm$import))
 (export "stackAlloc" (func $stackAlloc))
 (func $stackAlloc (; 2 ;) (type $FUNCSIG$v)
  (drop
   (global.get $STACKTOP)
  )
  (call $imported_twice_a)
 )
)
