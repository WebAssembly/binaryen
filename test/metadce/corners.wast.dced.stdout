unused: global$UNUSEDTOP$2
unused: ignorable import
